VideoLocker/src/main/java/org/edx/mobile/authentication/LoginAPI.java

```diff
@@ -104,13 +102,20 @@ public AuthResponse logInUsingGoogle(String accessToken) throws Exception {
     @NonNull
     private AuthResponse finishSocialLogIn(@NonNull String accessToken, @NonNull LoginPrefs.AuthBackend authBackend) throws Exception {
         final String backend = ApiConstants.getOAuthGroupIdForAuthBackend(authBackend);
-        final AuthResponse response = loginService.exchangeAccessToken(accessToken, config.getOAuthClientId(), backend);
-        if (response.error != null && response.error.equals("401")) {
+        final Response<AuthResponse> response = loginService.exchangeAccessToken(accessToken, config.getOAuthClientId(), backend).execute();
+        if (response.code() == HttpURLConnection.HTTP_UNAUTHORIZED) {
             // TODO: Introduce a more explicit error code to indicate that an account is not linked.
```

Comment:
> This seems to be a comment for the server-side implementation.

Yup, sadly there is no way to tell if the server is giving a generic "unauthorized error", or if it's because the account is not linked :cry: